### DISCUSSION TITLE
Include period PoP in forecast API

### DISCUSSION THREAD

**Discussion by oisact** [weather-gov/api#249] [Q&A]
The daily / period forecast API (eg https://api.weather.gov//gridpoints//RNK//34,49//forecast) does not include a discrete PoP (probability of precipitation) value.  It appears possible to jump through some hoops to extract a low-precision (10% increment) value from the forecast icon URL (as long as the value is 20% or greater).  However it is far more desirable to simply have PoP as a discrete value.

The probability of rain for a given day / period is a very fundamental part of a forecast, so it seems to be an oversight that it is not provided.

Note: This data is included in the hourly grid forecast, however it is not possible to calculate a daily PoP from a set of hourly PoP without knowing the forecast model or dependent variables.

**Comment by robstitt** [comment]
Note: I don't see it in the hourly forecast either (just a semi "free-form" description like "Chance Rain Showers". I agree, the POP % AND type really should be provided.

**Reply by JJKraw** [reply]
I believe that if the POP is above a certain threshold (I don't remember the number), it appears numerically in the detailedForecast.  E.g., "Chance of precipitation is 90%.".  So if you see "chance" and no number, it's below whatever that threshold is.

**Reply by scadergit** [reply]
As an internal developer on other weather.com apps, I agree, but there have been more critical priorities.

**Reply by oisact** [reply]
It is in the gridpoints API, in json format. It takes a little more effort to parse as it isn't hourly, but periods, using a sort of compression (the period is as long as the value is unchanged. IE if the temperature is 70 for 3 hours, then the period will cover 3 hours).
Example:
https://api.weather.gov/gridpoints/CLE/99,56

**Reply by robstitt** [reply]
Interesting. That data is WAY more complex--but has a lot of useful information. I'll have to ponder trying to figure it all out.

Adding a few values to the other API would be great--but this might offer an alternative in the mean time. Thx.

**Reply by scadergit** [reply]
The API forecast endpoint is simply a copy of the json returned by forecast.weather.gov to make the transition easier, but It has been a long standing goal to come back around and enhance the endpoint. We're just a very small team tasked with other priorities, even beyond the API itself (we support all weather.gov properties).

**Reply by robstitt** [reply]
I'm not the only one doing way more than 1 FTE's worth of work? 

Yeah, I know the feeling.
